docs/semgrep-code/java.md

```diff
@@ -0,0 +1,80 @@
+---
+slug: java
+append_help_link: true
+title: Semgrep Pro in Java
+hide_title: true
+description: How Semgrep Pro Engine can increase true positives and reduce false positives in the Java language.
+tags:
+  - tk
+---
+
+# Semgrep Pro performance in Java
+
+<!-- False positive reduction in Java through Semgrep Pro -->
+
+This document explains how Semgrep Pro Engine detects true positives and reduces false positives in Java.
+
+Additionally, it provides several simple rule examples to illustrate the concepts and how you can make use of these Semgrep features when writing your own rules.
+
+## Language features prevent injection through boolean and integer types 
+
+Strong typing in Java, combined with its compile-time and runtime checks, reduces the likelihood that an integer or boolean input will be exploited to perform injection-style attacks. Semgrep Pro can reduce false positives by leveraging these checks.
+
+The OSS Engine matches based on patterns, which can result in false positivies (FPs), but the Pro Engine can detect boolean and integer values and mark these as untainted, or safe, eliminating FPs.
+
+### Example: `int-bool-untainted`
+
+<iframe title="tk" src="https://semgrep.dev/embed/editor?snippet=Ab0p8" width="100%" height="432px" frameBorder="0"></iframe>
+
+**Figure**. `int-bool-untainted`. To view the entire sample code and rule, click **Open in Playground**.
+
+This demo rule detects tainted data in `sink`. 
+
+* This example has two true positives: **line 22** and **line 28**.
+* Semgrep Pro is able to detect that **line 24 and 30 are false positives**. Semgrep OSS can't catch that distinction.
+* Line 24 is a false positive because the data in the sink is an element of an integer List.
```

Comment:
typo: List -> list ?

Reply:
You could just say "it's an integer" too.

Reply:
Choosing to keep this as is, corrected List to list. TY.